src/Model/CollectionInfo.php

```diff
@@ -108,6 +108,30 @@ public function isCapped()
         return ! empty($this->info['options']['capped']);
     }
 
+    /**
+     * Return whether the collection is a collection.
+     */
+    public function isCollection() : bool
+    {
+        return $this->info['type'] == 'collection';
+    }
```

Comment:
I was considering this as well. I'm thinking this:
```php
interface CollectionInfo
{
    /** Returns the name of the collection or view */
    public function getName() : string;

    /** Returns info.type */
    public function getType() : string;

    /** Returns info.options */
    public function getOptions() : array;

    /** Returns the raw collection info response from listCollections */
    public function getRawInfo() : array;
}
```

I'm not sure what to do with the `isCapped` helper though, whether to keep or deprecate it.

A different, more involved approach, would see different classes for types:
```php
/** @final */
class Collection implements CollectionInfo
{
}

final class CappedCollection extends Collection
{
    public function getMax() : ?int {}

    public function getSize() : ?int {}
}

final class View implements CollectionInfo
{
}
```

Reply:
Giving this some more though: If I knew nothing about this class, I think my first impression of `getRawInfo` would be the method that returns _everything_ (like `__debugInfo`). I think `getInfo` makes more sense and we can just clearly document that it returns the "info" property of the server response.

> I'm not sure what to do with the isCapped helper though, whether to keep or deprecate it.

I think we can deprecate it in documentation. That would give us something to point to if an idea for other specific helpers comes up down the line.

> A different, more involved approach, would see different classes for the various types exposed through CollectionInfo:

While that would be very usable (folks could use `instanceof`), I don't think we have the data to invest in this API -- and that's basically taking us down a road of query builders where we'd need to maintain this for each server update. Perhaps worse because I don't think the server response is fully documented.

This does seem like something more fitting for an auxiliary library ([PHPLIB-4](https://jira.mongodb.org/browse/PHPLIB-4)), which is where we originally intended to build out APIs for other shell-like commands (e.g. user management, misc. helpers).

Reply:
Good point. I've added getters for `idIndex`, `info`, and `type` which previously didn't have any. I also added `@deprecated` phpdoc comments to the `isCapped`, `getCappedMax`, and `getCappedSize` methods, but they don't trigger any deprecations warnings yet.